index.js

```diff
@@ -0,0 +1,130 @@
+      }
+
+      const keys = Object.keys(obj)
+
+      if (keys.length > 0) {
+        // Recursive transform
+        const out = {}
+        keys.forEach((key) => {
+          if (typeof obj[key] === 'object') {
+            out[key] = transform(obj[key])
+          } else {
+            out[key] = obj[key]
+          }
+        })
+        return out
+      } else {
+        return obj
+      }
+    }
+
+    return transform(dagNode)
+  }
+
+  const defaultTags = {
+    [CID_CBOR_TAG]: (val) => {
+      // remove that 0
+      val = Uint8Array.from(val.slice(1))
+      const [version] = varint.decode(val)
+      if (version > 1) {
+        // CIDv0
+        return new CID(0, 0x70, val)
+      }
+      return new CID(val)
+    }
+  }
+  const defaultSize = 64 * 1024 // current decoder heap size, 64 Kb
+  let currentSize = defaultSize
+  const defaultMaxSize = 64 * 1024 * 1024 // max heap size when auto-growing, 64 Mb
+  let maxSize = defaultMaxSize
+  let decoder = null
+
+  /**
+   * Configure the underlying CBOR decoder.
+   *
+   * @param {Object} [options] - The options the decoder takes. The decoder will reset to the defaul values if no options are given.
+   * @param {number} [options.size=65536] - The current heap size used in CBOR parsing, this may grow automatically as larger blocks are encountered up to `maxSize`
+   * @param {number} [options.maxSize=67108864] - The maximum size the CBOR parsing heap is allowed to grow to before `dagCBOR.util.deserialize()` returns an error
+   * @param {Object} [options.tags] - An object whose keys are CBOR tag numbers and values are transform functions that accept a `value` and return a decoded representation of that `value`
+   */
+  const configureDecoder = (options) => {
+    const tags = defaultTags
+
+    if (options) {
+      if (typeof options.size === 'number') {
+        currentSize = options.size
+      }
+      if (typeof options.maxSize === 'number') {
+        maxSize = options.maxSize
+      }
+    } else {
+      // no options, reset to defaults
+      currentSize = defaultSize
+      maxSize = defaultMaxSize
+    }
+
+    const decoderOptions = {
+      tags,
+      size: currentSize
+    }
+
+    decoder = new cbor.Decoder(decoderOptions)
+    // borc edits opts.size in-place so we can capture _actual_ size
+    currentSize = decoderOptions.size
+  }
+  configureDecoder()
+  module.exports.configureDecoder = configureDecoder // for testing
+
+  const encode = (node) => {
+    const nodeTagged = replaceCIDbyTAG(node)
+    const serialized = cbor.encode(nodeTagged)
+    return serialized
+  }
+
+  const decode = (data) => {
+    if (data.length > currentSize && data.length <= maxSize) {
+      configureDecoder({ size: data.length })
+    }
+
+    if (data.length > currentSize) {
+      throw new Error('Data is too large to deserialize with current decoder')
+    }
+
+    const deserialized = decoder.decodeFirst(data)
+    return deserialized
+  }
+
+  const code = 0x71
```

Comment:
tiny nit: could we pull these two to the top level and at the top of the file?

Reply:
good catch, i agree.